src/main/java/org/jenkinsci/plugins/workflow/libs/SCMSourceRetriever.java

```diff
@@ -75,13 +75,13 @@ public SCMSource getScm() {
         if (revision == null) {
             throw new AbortException("No version " + version + " found for library " + name);
         }
-        doRetrieve(scm.build(revision.getHead(), revision), revision.isDeterministic(), target, run, listener);
+        doRetrieve(scm.build(revision.getHead(), revision), target, run, listener);
     }
 
-    static void doRetrieve(@Nonnull SCM scm, boolean deterministic, FilePath target, Run<?, ?> run, TaskListener listener) throws Exception {
+    static void doRetrieve(@Nonnull SCM scm, FilePath target, Run<?, ?> run, TaskListener listener) throws Exception {
         // Adapted from CpsScmFlowDefinition:
         SCMStep delegate = new GenericSCMStep(scm);
-        delegate.setPoll(!deterministic); // TODO is this desirable?
+        delegate.setPoll(false); // TODO we have no API for determining if a given SCMHead is branch-like or tag-like; would we want to turn on polling if the former?
```

Comment:
IMHO The library is not the one being built, it is just a tool to achieve the build. So for me it is ok to never poll the library.
If I want to test the library I'll setup a separate job for that :)